### DISCUSSION TITLE
How to handle different `flakeDir` and `workDir` in `om.ci`?

### DISCUSSION THREAD

**Discussion by shivaraj-bh** [juspay/omnix#284] [Q&A] [answered]
```nix
{
  om.ci.default = {
    root = {
      dir = ".";
      steps = {
        custom = {
          cz-check = {
            type = "app";
            name = "cz-check";
          };
        };
      };
    };
  };
}
```
source: https://github.com/juspay/services-flake/pull/338

In this config, `cz-check` is an app present in the `dev` sub-flake whereas the `workDir` is supposed to be the project root.

**Comment by srid** [comment, marked as answer]
Can't that custom app step be put in the `dev` (rather than root) subflake? Since for `git`, $PWD is irrelevant, as it will traverse parent directories.
